presto-main/src/main/java/com/facebook/presto/type/TypeRegistry.java

```diff
@@ -278,9 +286,21 @@ private Type getCommonSuperTypeForDecimal(DecimalType firstType, DecimalType sec
         return createDecimalType(targetPrecision, targetScale);
     }
 
-    private Type getCommonSuperTypeForVarchar(VarcharType firstType, VarcharType secondType)
+    private Type getCommonSuperTypeForCharacterStrings(Type firstType, Type secondType)
+    {
+        int returnTypeLength = Math.max(getCharacterStringTypeLength(firstType), getCharacterStringTypeLength(secondType));
+        if (firstType instanceof VarcharType || secondType instanceof VarcharType) {
+            return createVarcharType(returnTypeLength);
+        }
+        return createCharType(returnTypeLength);
```

Comment:
I think it also depends where the data comes from and what is the target:
`When a character string is taken from SQL-data to be assigned to a fixed length` -> truncate other than spaces triggers warning
`When a character string is assigned to a fixed length SQL-data character string` -> truncate triggers error.

Either way, to be consistent with VARCHAR there should be no truncate.